scio-core/src/main/scala/com/spotify/scio/coders/instances/JavaCoders.scala

```diff
@@ -98,6 +98,8 @@ trait JavaCoders extends JavaBeanCoders {
   implicit val jDoubleCoder: Coder[java.lang.Double] = fromScalaCoder(Coder.doubleCoder)
 
   implicit val jBooleanCoder: Coder[java.lang.Boolean] = Coder.beam(BooleanCoder.of())
+  implicit val jRemoteFileUtil: Coder[RemoteFileUtil] =
+    Coder.beam(SerializableCoder.of(classOf[RemoteFileUtil]))
```

Comment:
I've very suspicious about this.
IMHO `RemoteFileUtil` should not have a coder. It implements serializable to be placed in the graph as part of DoFn closure (And I really think we should not even create an instance and make it pure static).
`RemoteFileUtil` should not be part of any data, thus should not require a coder

Reply:
yeah -- I was looking for a quick win on Coders without having to change any signatures so MiMA would be happy, but I think I'll need to do some refactoring on RFU/SparkeyUri so we don't have to hack around them like this

Reply:
Can you point where the `RemoteFileUtil` might be part of the data ?
Even without refactoring, Looks to me we can live without a `Coder[RemoteFileUtil]`, right ?

Reply:
a few places --

- [FileDownloadFn](https://github.com/spotify/scio/blob/main/scio-core/src/main/java/com/spotify/scio/transforms/FileDownloadDoFn.java#L41) as instance variable
- [DistCache](https://github.com/spotify/scio/blob/main/scio-core/src/main/scala/com/spotify/scio/values/DistCache.scala#L45) (I need to double-check on this one how DistCache gets serialized; maybe RFU is not part of data)
- [RemoteSparkeyUri](https://github.com/spotify/scio/blob/main/scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/SparkeyUri.scala#L71) which is the return type of one of the sparkey side-input views

I think we can definitely redesign this as a completely static class (scala Object style) that doesn't get serialized into the job graph -- will try to work on that today/tomorrow.